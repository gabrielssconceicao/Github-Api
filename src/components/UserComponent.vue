<script setup lang="ts">
import type { IUser } from '@/types/user';
import type { RouterLink } from 'vue-router';
const props = defineProps<{ user: IUser }>()
</script>
<template>
  <section class="h-[calc(100dvh-149.5px)] md:h-[calc(100vh-88px)] bg-section flex justify-center items-center p-4  ">
    <div class="
      container max-w-[500px]
      flex flex-col items-center justify-between gap-3
    bg-white p-3 rounded-lg shadow-md shadow-[#00000088]
      relative
      after:absolute after:h-2 after:w-full after:bottom-[0] after:rounded-b-lg after:bg-primary-contrast ">
      <div class="flex flex-col items-center justify-between gap-3" id="user-info
      ">
        <img :src="props.user.avatar_url" :alt="`${props.user.login} avatar`"
          class="rounded-full w-28 border-4 border-primary" aria-label="avatar" />
        <h2 v-if="props.user.name" class="font-bold font-sans text-xl tracking-wider" aria-label="user-name">{{
          props.user.name }}
        </h2>
        <a class="font-mono font-semibold text-lg bg-primary-contrast flex items-center gap-2 p-1 rounded-md"
          :href="props.user.html_url" target="_blank">
          <img src="@/assets/images/github-logo.svg" alt="github-logo" class="w-6" aria-label="github-logo">
          <span class="text-white" aria-label="github-url">{{ props.user.login }}</span></a>
        <p v-if="props.user.bio" class="font-mono break-before-auto" aria-label="bio">{{ props.user.bio }}</p>
      </div>

      <div class="user-stats flex flex-col gap-3 w-[50%]">
        <div class="follows flex justify-around gap-2 flex-1">
          <p class=" stat" aria-label="followers">Followers <span>{{ props.user.followers }}</span></p>
          <p class="stat" aria-label="following">Following <span>{{ props.user.following }}</span></p>
        </div>
        <div class="flex justify-around gap-2 flex-1">
        </div>
        <p class="stat flex-1" v-if="props.user.location" aria-label="location">Location
          <span>{{ props.user.location }}</span>
        </p>
      </div>

      <RouterLink to="/repos"
        class="bg-primary p-2 w-full text-center text-white hover:bg-primary-contrast font-mono rounded-md">View
        Repositories</RouterLink>

    </div>


  </section>
</template>
