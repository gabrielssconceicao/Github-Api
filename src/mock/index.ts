import { type IUser } from '@/types/user';
import { type IRepos } from '@/types/repos';

export const user: IUser = {
  name: 'Teste User',
  login: 'testeuser',
  avatar_url: 'https://avatars.githubusercontent.com/u/10590076?v=4',
  followers: 0,
  following: 5,
  location: 'Brazil',
  html_url: 'https://github.com/testeuser',
  bio: 'lorem ipsum dolor sit amet',
  public_repos: 5,
};
// JS - 3
// HTML - 2
// python - 1
// Ruby - 1
//Java - 1
//C++ - 1
//Typescript - 1
export const repos: IRepos[] = [
  {
    name: 'Teste Repo 1',
    description: 'lorem ipsum dolor sit amet',
    html_url: 'https://github.com/testeuser/testerepo',
    language: 'Javascript',
  },
  {
    name: 'Teste Repo 2',
    description: 'lorem ipsum dolor sit amet',
    html_url: 'https://github.com/testeuser/testerepo',
    language: 'HTML',
  },
  {
    name: 'Teste Repo 3',
    description: 'lorem ipsum dolor sit amet',
    html_url: 'https://github.com/testeuser/testerepo',
    language: 'Python',
  },
  {
    name: 'Teste Repo 4',
    description: 'lorem ipsum dolor sit amet',
    html_url: 'https://github.com/testeuser/testerepo',
    language: 'Javascript',
  },
  {
    name: 'Teste Repo 5',
    description: 'lorem ipsum dolor sit amet',
    html_url: 'https://github.com/testeuser/testerepo',
    language: 'Javascript',
  },
  {
    name: 'Teste Repo 6',
    description: 'lorem ipsum dolor sit amet',
    html_url: 'https://github.com/testeuser/testerepo',
    language: 'HTML',
  },
  {
    name: 'Teste Repo 7',
    description: 'lorem ipsum dolor sit amet',
    html_url: 'https://github.com/testeuser/testerepo',
    language: 'Ruby',
  },
  {
    name: 'Teste Repo 8',
    description: 'lorem ipsum dolor sit amet',
    html_url: 'https://github.com/testeuser/testerepo',
    language: 'Java',
  },
  {
    name: 'Teste Repo 9',
    description: 'lorem ipsum dolor sit amet',
    html_url: 'https://github.com/testeuser/testerepo',
    language: 'C++',
  },
  {
    name: 'Teste Repo 10',
    description: 'lorem ipsum dolor sit amet',
    html_url: 'https://github.com/testeuser/testerepo',
    language: 'Typescript',
  },
];
