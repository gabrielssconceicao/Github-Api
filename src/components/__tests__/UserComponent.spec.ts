import { describe, it, expect } from 'vitest';
import { mount } from '@vue/test-utils';
import UserComponent from '../UserComponent.vue';
import { user as userMock } from '@/mock';

describe('<UserComponent />', () => {
  it('should render correctly', () => {
    const wrapper = mount(UserComponent, {
      props: {
        user: { ...userMock },
      },
    });

    expect(wrapper.find('section').exists()).toBe(true);

    const userAvatar = wrapper.find('img[aria-label="avatar"]');
    expect(userAvatar.exists()).toBe(true);
    expect(userAvatar.attributes('src')).toBe(userMock.avatar_url);
    expect(userAvatar.attributes('alt')).toBe(`${userMock.login} avatar`);

    const userName = wrapper.find('h2');
    expect(userName.exists()).toBe(true);
    expect(userName.text()).toBe(userMock.name);

    const userLink = wrapper.find('a[aria-label="user-github-link"]');
    expect(userLink.exists()).toBe(true);
    expect(userLink.attributes('href')).toBe(userMock.html_url);
    const userLogin = userLink.find('span[aria-label="user-github-name"]');
    expect(userLogin.exists()).toBe(true);
    expect(userLogin.text()).toBe(userMock.login);

    const userBio = wrapper.find('p[aria-label="user-bio"');
    expect(userBio.exists()).toBe(true);
    expect(userBio.text()).toBe(userMock.bio);

    const userFollowers = wrapper.find('p[aria-label="user-followers"]');
    expect(userFollowers.exists()).toBe(true);
    expect(userFollowers.text()).toBe(`Followers ${userMock.followers}`);

    const userFollowing = wrapper.find('p[aria-label="user-following"]');
    expect(userFollowing.exists()).toBe(true);
    expect(userFollowing.text()).toBe(`Following ${userMock.following}`);

    const userRepos = wrapper.find('p[aria-label="user-repos"]');
    expect(userRepos.exists()).toBe(true);
    expect(userRepos.text()).toBe(`Public Repos ${userMock.public_repos}`);

    const userReposLink = wrapper.find('[aria-label="user-repos-link"]');
    expect(userReposLink.exists()).toBe(true);
  });

  it('should render correctly with null props', () => {
    const wrapper = mount(UserComponent, {
      props: {
        user: {
          ...userMock,
          name: null,
          location: null,
          bio: null,
        },
      },
    });

    const userName = wrapper.find('h2');
    expect(userName.exists()).toBe(false);

    const userBio = wrapper.find('p[aria-label="user-bio"');
    expect(userBio.exists()).toBe(false);

    const userFollowers = wrapper.find('p[aria-label="user-followers"]');
    expect(userFollowers.exists()).toBe(true);
    expect(userFollowers.text()).toBe(`Followers ${userMock.followers}`);

    const userFollowing = wrapper.find('p[aria-label="user-following"]');
    expect(userFollowing.exists()).toBe(true);
    expect(userFollowing.text()).toBe(`Following ${userMock.following}`);

    const userLink = wrapper.find('a[aria-label="user-github-link"]');
    expect(userLink.exists()).toBe(true);
    expect(userLink.attributes('href')).toBe(userMock.html_url);
    const userLogin = userLink.find('span[aria-label="user-github-name"]');
    expect(userLogin.exists()).toBe(true);
    expect(userLogin.text()).toBe(userMock.login);

    const userAvatar = wrapper.find('img[aria-label="avatar"]');
    expect(userAvatar.exists()).toBe(true);
    expect(userAvatar.attributes('src')).toBe(userMock.avatar_url);
    expect(userAvatar.attributes('alt')).toBe(`${userMock.login} avatar`);
  });

  it('should match snapshot', () => {
    const wrapper = mount(UserComponent, {
      props: {
        user: { ...userMock },
      },
    });
    expect(wrapper.element).toMatchSnapshot();
  });
});
