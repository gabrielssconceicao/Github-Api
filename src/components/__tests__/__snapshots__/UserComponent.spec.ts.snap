// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<UserComponent /> > should match snapshot 1`] = `
<section
  class="h-[calc(100dvh-149.5px)] md:h-[calc(100vh-88px)] bg-section flex justify-center items-center p-4"
>
  <div
    class="container max-w-[500px] flex flex-col items-center justify-between gap-3 bg-white p-3 rounded-lg shadow-md shadow-[#00000088] relative after:absolute after:h-2 after:w-full after:bottom-[0] after:rounded-b-lg after:bg-primary-contrast"
  >
    <div
      class="flex flex-col items-center justify-between gap-3"
      id="user-info
      "
    >
      <img
        alt="testeuser avatar"
        aria-label="avatar"
        class="rounded-full w-28 border-4 border-primary"
        src="https://avatars.githubusercontent.com/u/10590076?v=4"
      />
      <h2
        aria-label="user-name"
        class="font-bold font-sans text-xl tracking-wider"
      >
        Teste User
      </h2>
      <a
        aria-label="user-github-link"
        class="font-mono font-semibold text-lg bg-primary-contrast flex items-center gap-2 p-1 rounded-md"
        href="https://github.com/testeuser"
        target="_blank"
      >
        <img
          alt="github-logo"
          aria-label="github-logo"
          class="w-6"
          src="/src/assets/images/github-logo.svg"
        />
        <span
          aria-label="user-github-name"
          class="text-white"
        >
          testeuser
        </span>
      </a>
      <p
        aria-label="user-bio"
        class="font-mono break-before-auto"
      >
        lorem ipsum dolor sit amet
      </p>
    </div>
    <div
      class="user-stats flex flex-col gap-3 w-[50%]"
    >
      <div
        class="follows flex justify-around gap-2 flex-1"
      >
        <p
          aria-label="user-followers"
          class="stat"
        >
          Followers 
          <span>
            0
          </span>
        </p>
        <p
          aria-label="user-following"
          class="stat"
        >
          Following 
          <span>
            5
          </span>
        </p>
      </div>
      <div
        class="flex justify-around gap-2 flex-1"
      />
      <p
        aria-label="user-location"
        class="stat flex-1"
      >
        Location 
        <span>
          Brazil
        </span>
      </p>
    </div>
    <routerlink
      class="bg-primary p-2 w-full text-center text-white hover:bg-primary-contrast font-mono rounded-md"
      to="/repos"
    >
      View Repositories
    </routerlink>
  </div>
</section>
`;
